src/lib/esnext.promise.d.ts

```diff
@@ -0,0 +1,12 @@
+/**
+ * Represents the completion of an asynchronous operation
+ */
+interface Promise<T> {
+    /**
+     * Attaches a callback that is invoked when the Promise is settled (fulfilled or rejected). The
+     * resolved value cannot be modified from the callback.
+     * @param onfinally The callback to execute when the Promise is settled (fulfilled or rejected).
+     * @returns A Promise for the completion of the callback.
+     */
+    finally(onfinally?: (() => void) | undefined | null): Promise<T>

```

Comment:
Would a return type of `this` be appropriate here, or better to keep `Promise<T>`?

It doesn't return the receiver itself; but the returned value has the same type as the receiver...... if the receiver implements `[Symbol.species]`. This is probably out of scope of what the type system can do...